### PR TITLE
Added bool type support for jsonpath.

### LINKED ISSUE #28678: kubectl JSONPath doesn't allow use of `true` or `false` keywords
`kubectl` raises an error if you use the `true`/`false` keywords in a filter. e.g.,

``` bash
$ kubectl get pods --all-namespaces -o 'jsonpath={.items[*].status.containerStatuses[?(@.ready==false)].name}'

Error executing template: unrecognized identifier false. Printing more information for debugging the template:
    template was:
        {.items[*].status.containerStatuses[?(@.ready==false)].name}
...
error executing jsonpath "{.items[*].status.containerStatuses[?(@.ready==false)].name}": unrecognized identifier false
```

You also can't use the `!` operator (e.g., `?(!@.ready)`), so I can't see any way to check if a property is false.

I've never used JSONPath before this, but it appears to be a valid query when used on http://jsonpath.com/ (although I can't find any reference documentation for JSONPath filters, so I'm not sure).

### PR DESCRIPTION
**Pull request by xingzhou**
Added keywords "true" and "false" for
supporting bool type in jsonpath.

Fixed #28678

### CODE REVIEW

File: pkg/util/jsonpath/parser.go [hunk 1]
```diff
@@ -194,7 +194,18 @@ func (p *Parser) parseIdentifier(cur *ListNode) error {
 		}
 	}
 	value := p.consumeText()
-	cur.append(newIdentifier(value))
+
+	if isBool(value) {
+		v, err := strconv.ParseBool(value)
+		if err != nil {
+			return fmt.Errorf("can not parse bool %s", value)
```
Comment (brendandburns) [inline pkg/util/jsonpath/parser.go]: Please include the `err.Error()` string in here too, for easier debugging.

**Comment by xingzhou** [comment]
Thanks @brendandburns for the review. I've updated the code according to your comments, please take a look.

**Comment by brendandburns** [comment]
@k8s-bot gci gce e2e test this

**Comment by brendandburns** [comment]
LGTM.

thanks again.

**Comment by xingzhou** [comment]
Thanks @brendandburns, Jenkins kops AWS e2e test is hang, shall we re-kick it?

**Comment by brendandburns** [comment]
@justinsb do you know how to kick the AWS e2e bot?

Thanks

**Comment by xingzhou** [comment]
Just rebased the patch to the latest code, @brendandburns would you please help re-add the lgtm tag? hoping this can re-kick all the tests and merge the patch, thank you

**Comment by xingzhou** [comment]
@brendandburns, have rebased the code, would you please help to review it? thanks

**Comment by wojtek-t** [comment]
@xingzhou - can you please rebase one more time?

**Comment by xingzhou** [comment]
@wojtek-t, thanks for the reminder, have rebased the code, please take a look.

**Comment by xingzhou** [comment]
@k8s-bot unit test this

**Pull request merged**